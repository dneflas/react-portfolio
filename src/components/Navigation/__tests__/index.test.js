import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "..";

const mockCurrentSection = jest.fn();
const mockSetCurrentSection = jest.fn();

afterEach(cleanup);

describe("Nav components render", () => {
  it("renders", () => {
    render(
      <Nav
        currentSection={mockCurrentSection}
        setCurrentSection={mockSetCurrentSection}
      />
    );
  });
  it("matches snapshot", () => {
    const { asFragment } = render(
      <Nav
        currentSection={mockCurrentSection}
        setCurrentSection={mockSetCurrentSection}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Links are visible", () => {
  it("inserts text into links", () => {
    const { getByTestId } = render(
      <Nav
        currentSection={mockCurrentSection}
        setCurrentSection={mockSetCurrentSection}
      />
    );
    expect(getByTestId("about")).toHaveTextContent("About Me");
    expect(getByTestId("portfolio")).toHaveTextContent("Portfolio");
    expect(getByTestId("contact")).toHaveTextContent("Contact");
    expect(getByTestId("resume")).toHaveTextContent("Resume");
  });
});
