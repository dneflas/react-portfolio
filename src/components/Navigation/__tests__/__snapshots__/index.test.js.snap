// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav components render matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="flex-row"
  >
    <a
      class="ml-2 my-1 px-2 mx-5 py-1 text-dark false"
      data-testid="about"
      href="#about"
    >
      About Me
    </a>
    <a
      class="ml-2 my-1 px-2 mx-5 py-1 text-dark false"
      data-testid="portfolio"
      href="#portfolio"
    >
      Portfolio
    </a>
    <a
      class="ml-2 my-1 px-2 mx-5 py-1 text-dark false"
      data-testid="contact"
      href="#contact"
    >
      Contact
    </a>
    <a
      class="ml-2 my-1 px-2 mx-5 py-1 text-dark false"
      data-testid="resume"
      href="#resume"
    >
      Resume
    </a>
  </nav>
</DocumentFragment>
`;
